{
 "metadata": {
  "name": "",
  "signature": "sha256:2ec9cbb85e119bd9b95db35a8113e9750c02658ab477f9aa729fdf5948fe2f7b"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import pandas as pd\n",
      "import numpy as np\n",
      "import glob, os\n",
      "import subprocess\n",
      "#os.chdir('PDFs')\n",
      "import re\n",
      "from skbio.alignment import global_pairwise_align"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 3
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import distance"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 4
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "os.getcwd()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 7,
       "text": [
        "'/home/bonobo/Documents/GMO/PDFs'"
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "os.chdir('PDFs')"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 6
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import pyPdf\n",
      " \n",
      "# parent = \"C:/Users/victoryee/Google Drive/Projects/extract-pdf-text\"\n",
      "# os.chdir(parent)\n",
      "\n",
      "# print getPDFContent(filename).encode(\"ascii\", \"xmlcharrefreplace\")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 8
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def getPDFContent(path):\n",
      "    content = \"\"\n",
      "    # Load PDF into pyPDF\n",
      "    pdf = pyPdf.PdfFileReader(file(path, \"rb\"))\n",
      "    # Iterate pages\n",
      "    for i in range(0, pdf.getNumPages()):\n",
      "        # Extract text from page and add to content\n",
      "        content += pdf.getPage(i).extractText() + \"/n\"\n",
      "    # Collapse whitespace\n",
      "    content = \" \".join(content.replace(u\"/xa0\", \" \").strip().split())\n",
      "    return content"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 9
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "filelist = {}\n",
      "\n",
      "filename_list = ['1.pdfx', '2.pdfx', '3.pdfx']\n",
      "\n",
      "for files in filename_list:\n",
      "    filename = os.path.abspath(files)\n",
      "    \n",
      "    filelist[files] = getPDFContent(filename).encode(\"ascii\", \"ignore\")\n",
      "    filelist[files] = filelist[files].split('References')[1]\n",
      "\n",
      "    m = re.findall(r'\\([0-9]{4}\\)(.+?)(?:http\\:|\\([0-9]+|(?:\\:[0-9]+?))', filelist[files])\n",
      "\n",
      "    new_list = []\n",
      "\n",
      "    for i in range(len(m)):\n",
      "        found = re.search(r'^([^.]+).*', m[i])\n",
      "        if found:\n",
      "            if len(found.group(1)) > 10:\n",
      "                string = found.group(1)\n",
      "                string = ''.join(e for e in string if e.isalnum())\n",
      "                new_list.append(string)\n",
      "    filelist[files] = new_list"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 10
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def compare_dist(first, second):\n",
      "    return distance.levenshtein(first, second, normalized=True)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 11
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len(filelist)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 12,
       "text": [
        "3"
       ]
      }
     ],
     "prompt_number": 12
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "dist_matrix = []\n",
      "#need to fix\n",
      "for files in range(len(filelist)):\n",
      "    for i in range(len(filelist[\"1.pdfx\"])):\n",
      "        row = []\n",
      "        for j in range(len(filelist[\"2.pdfx\"])):\n",
      "            row.append(1-compare_dist(filelist[\"1.pdfx\"][i], filelist[\"2.pdfx\"][j]))\n",
      "        dist_matrix.append(row)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 13
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "df = pd.DataFrame(dist_matrix)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 18
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "filelist"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 16,
       "text": [
        "{'1.pdfx': ['Enhancedfitnessduetohigherfecundityincreaseddefenceagainstaspecialistandtolerancetowardsageneralistherbivoreinaninvasiveannualplant',\n",
        "  'CroptoweedgeneflowinthegenusSorghumPoaceaespontaneousinterspecifichybridizationbetweenJohnsongrassSorghumhalepenseandcropsorghumS',\n",
        "  'Cropferalityimplicationsfornoveltraitconfinement',\n",
        "  'ThebiologyandecologyofferalalfalfaMedicagosativaL',\n",
        "  'Theimportanceofpopulationgrowthseeddispersalandhabitatsuitabilityindeterminingplantinvasiveness',\n",
        "  'Survivalandagronomicperformanceof25alfalfacultivarsandstrainsinterseededintorangeland',\n",
        "  'Worlddistributionandhistoricaldevelopments',\n",
        "  'Seedlingrecruitmentinanortherntemperateforesttherelativeimportanceofsupplyandestablishmentlimitation',\n",
        "  'DecisiondocumentDD200553determinationofthesafetyofMonsantoCanadaInc',\n",
        "  'Theeffectsofrateandmethodofseedingontheyieldandbotanicalcompositionofalfalfaorchardgrassandalfalfatallfescue',\n",
        "  'WhichtraitspromotepersistenceofferalGMcropsPart1implicationsofenvironmentalstochasticity',\n",
        "  'SeedlimitationandthedynamicsofferaloilseedrapeontheM25motorway',\n",
        "  'Plantsurvivalandrootandbudcompositionofalfalfapopulationsselectedforcontrastingfalldormancy',\n",
        "  'Themoleculargeneticsofcropdomestication',\n",
        "  'Variationinresourceacquisitionandutilizationtraitsbetweennativeandinvasiveperennialforbs',\n",
        "  'Competitionbetweengrassandlegumespeciesondryland',\n",
        "  'Geneticengineeringandpollenflow',\n",
        "  'BethedgingviaseedbankingindeserteveningprimosesOenotheraOnagraceaedemographicevidencefromnaturalpopulations',\n",
        "  'Environmentalphysiologyandcropgrowth',\n",
        "  'Resourceuseefficiencyandplantinvasioninlowresourcesystems',\n",
        "  'Lifespancorrelateswithpopulationdynamicsinperennialherbaceousplants',\n",
        "  'Usingaspatialandstagestructuredinvasionmodeltoassessthespreadofferalpopulationsoftransgenicoilseedrape',\n",
        "  'Acomparisonbetweencropdomesticationclassicalplantbreedingandgeneticengineering',\n",
        "  'Reducingtransgeneescaperoutes',\n",
        "  'Cropferalityandvolunteerism',\n",
        "  'ProlongedclonalgrowthescaperouteorroutetoextinctionOikos108',\n",
        "  'Hydraulicliftapotentiallyimportantecosystemprocess',\n",
        "  'Biochemicaltestforviability',\n",
        "  'EvidenceforgeneflowbetweenwildandcultivatedMedicagosativaLeguminosaebasedonallozymemarkersandquantitativetraits',\n",
        "  'Reseedingoldalfalfastands',\n",
        "  'ExperimentaldeterminationofseedbankagestructureinthewinterannualCollinsiaverna',\n",
        "  'BiogeographicsurveyofferalalfalfapopulationsintheU',\n",
        "  'Persistenceofalfalfainmixturewithgrassesinasemiaridregion',\n",
        "  'Geneflowandmultipleherbicideresistanceinescapedcanolapopulations',\n",
        "  'SASsystemformixedmodels',\n",
        "  'Tablesforanapproximatetestforoutliersinlinearmodels',\n",
        "  'EffectsofnutrientandCO2availabilityontolerancetoherbivoryinBrassicarapa',\n",
        "  'Lygusbugsonseedalfalfa',\n",
        "  'MunicipalcommunityProfiles',\n",
        "  'Varietyandyielddata',\n",
        "  'Foragequalityofglyphosateresistantalfalfaasinfluencedbyestablishmentsystems',\n",
        "  'Yieldmaturationandforagequalityofalfalfainablackwalnutalleycroppingpractice',\n",
        "  'Improvinggerminationofeveningprimroseseed',\n",
        "  'RangestudytourintheSovietUnion',\n",
        "  'Alfalfaseedproductionstudies',\n",
        "  'Allelopathy2ndedn',\n",
        "  'Evolutionofreproductiveeffortinametapopulationwithlocalextinctionsandecologicalsuccession',\n",
        "  'Reseedingbyeightalfalfapopulationsinasemiaridpasture',\n",
        "  'Survivalofalfalfainfivesemiaridrangeseedings',\n",
        "  'InsectpestsoflegumeandgrasscropsinwesternCanada',\n",
        "  'Availabilitydeterminationofnonregulatedstatusforalfalfageneticallyengineeredfortolerancetotheherbicideglyphosate',\n",
        "  'ThepotentialforthecoexistenceofGMandnonGMcropsinCanada',\n",
        "  'Evaluationofglyphosatetolerantandconventionalalfalfaweedcontrolsystemsduringthefirstyearofestablishment',\n",
        "  'Longevityofalfalfaseed',\n",
        "  'Biotechnologyandecoogytheecologicalrisksandbenefitsofgeneticallyengineeredplants',\n",
        "  'Recruitmentratesexhibithighelasticityandhightemporalvariationinpopulationsofashortlivedperennialherb',\n",
        "  'TransgenicoilseedrapealongtransportationroutersandportofVancouverinwesternCanada'],\n",
        " '2.pdfx': ['LocaldifferentiationinPlasmodiumfalciparumdrugresistancegenesinSudan',\n",
        "  'Genotypicvariationinplantdiseaseresistancephysiologicalresistanceinrelationtofielddiseasetransmission',\n",
        "  'GeographicaldistributionofselectedandputativelyneutralSNPsinSoutheastAsianmalariaparasites',\n",
        "  'IsolationandcharacterizationofmicrosatellitelocifromtherustpathogenMelampsoralini',\n",
        "  'FrequencyandfidelityofalienchromosometransmissioninGossypiumhexaploidbridgingpopulations',\n",
        "  'Geneticsofdiseaseresistanceinsheepandgoats',\n",
        "  'Hostpopulationsubdivisionandthegeneticstructureofnaturalpathogenpopulations',\n",
        "  'HostpathogeninteractionsinnaturalpopulationsofLinummarginaleandMelampsoralini',\n",
        "  'ThepopulationgeneticstructureoftherustfungusMelampsoraliniasrevealedbypathogenicityisozymeandRFLPmarkers',\n",
        "  'ChangedpatternsofresistanceinapopulationofLinummarginaleattackedbytherustpathogenMelampsoralini',\n",
        "  'CoevolutionarypatternsintheLinummarginaleMelampsoraliniassociationatacontinentalscale',\n",
        "  'Theecologicalgeneticsofhostparasiterelationships',\n",
        "  'TheRedQueenHypothesisandplantpathogeninteractions',\n",
        "  'Geneticsofsusceptibitlitytohumaninfectiousdisease',\n",
        "  'MultilocuspopulationstructureofTapesiayallundaeinWashingtonState',\n",
        "  'LoweffectivedispersalofasexualgenotypesinheterogeneouslandscapesbytheendemicpathogenPenicilliummarneffei',\n",
        "  'Emergenceofanewdiseaseasaresultofinterspecificvirulencegenetransfer',\n",
        "  'Asynthesisofexperimentalworkonparasitelocaladaptation',\n",
        "  'Diseaseandevolution',\n",
        "  'Sexualreproductionandtheevolutionofmicrobialpathogens',\n",
        "  'RatesofevolutionofavirulencephenotypesandDNAmarkersinanorthwestEuropeanpopulationofPucciniastriiformisf',\n",
        "  'HostpathogeninteractionsinnaturalpopulationsofLinummarginaleandMelampsoralini',\n",
        "  'Aspatialmechanismfortheevolutionandmaintenanceofsexualreproduction',\n",
        "  'Largescalespatialdynamicsofaspecialistplantpathogeninafragmentedlandscape',\n",
        "  'Distributionofgenefrequencyasatestofthetheoryoftheselectiveneutralityofpolymorphisms',\n",
        "  'Hostsexandlocaladaptationbyparasitesinasnailtrematodeinteraction',\n",
        "  'Thedetectionofdiseaseclusteringandageneralisedregressionapproach',\n",
        "  'Pathogenpopulationgeneticsevolutionarypotentialanddurableresistance',\n",
        "  'genotypeandgenodivetwoprogramsfortheanalysisofgeneticdiversityofasexualorganisms',\n",
        "  'Recombinationandthemultilocusstructureoffungalpopulations',\n",
        "  'MolecularEvolutionaryGenetics',\n",
        "  'Neutralgeneticdiversityinametapopulationwithrecurrentlocalextinctionandrecolonization',\n",
        "  'GenAlExv6GeneticAnalysisinExcel',\n",
        "  'NTSYSPcNumericalTaxonomyandMultivariateAnalysisSystemversion2',\n",
        "  'Neutraldriftandpolymorphismingeneforgenesystems',\n",
        "  'ComparitivestudiesofthepoplarrustsMelampsoramedusaeM',\n",
        "  'TheGeographicMosaicofCoevolution',\n",
        "  'Evolutionofvirulenceinaplanthostpathogenmetapopulation',\n",
        "  'LocaladaptationintheLinummarginaleMelampsoralinihostpathogeninteraction',\n",
        "  'VariationforneutralmarkersiscorrelatedwithvariationforquantitativetraitsintheplantpathogenicfungusMycosphaerellagraminicola'],\n",
        " '3.pdfx': ['Widespreaddistributionofproteorhodopsinsinfreshwaterandbrackishecosystems',\n",
        "  'Xanthorhodopsinaprotonpumpwithalightharvestingcarotenoidantenna',\n",
        "  'Farredradiationreectedfromadjacentleavesanearlysignalofcompetitioninplantcanopies',\n",
        "  'Bacterialcolonizationofleavesaspectrumofstrategies',\n",
        "  'Bacterialrhodopsinevidenceforanewtypeofphototrophyinthesea',\n",
        "  'Proteorhodopsinphototrophyintheocean',\n",
        "  'Communityproteogenomicsrevealsinsightsintothephysiologyofphyllospherebacteria',\n",
        "  'Primaryproductionofthebiosphereintegratingterrestrialandoceaniccomponents',\n",
        "  'Greenlightasignaltoslowdownorstop',\n",
        "  'Quantitativemetagenomicanalysesbasedonaveragegenomesizenormalization',\n",
        "  'ProteorhodopsinlateralgenetransferbetweenmarineplanktonicBacteriaandArchaea',\n",
        "  'ProteorhodopsinintheubiquitousmarinebacteriumSAR11',\n",
        "  'LightstimulatesgrowthofproteorhodopsincontainingmarineFlavobacteria',\n",
        "  'Proteorhodopsinphototrophyconfersenhancedsurvivalofmarinebacteriaduringstarvation',\n",
        "  'AbundantanddiversebacteriainvolvedinDMSPdegradationinmarinesurfacewaters',\n",
        "  'Dendroscopeaninteractiveviewerforlargephylogenetictrees',\n",
        "  'Upcomingchallengesformultiplesequencealignmentmethodsinthehighthroughputera',\n",
        "  'ProteorhodopsinbearingbacteriainAntarcticseaice',\n",
        "  'Plantassociatedmethylobacteriaascoevolvedphytosymbiontsahypothesis',\n",
        "  'Xanthorhodopsinabacteriorhodopsinlikeprotonpumpwithacarotenoidantenna',\n",
        "  'Microbiologyofthephyllosphere',\n",
        "  'Mesquiteamodularsystemforevolutionaryanalysis',\n",
        "  'Diversicationandspectraltuninginmarineproteorhodopsins',\n",
        "  'ThemetagenomicsRASTserverapublicresourcefortheautomaticphylogeneticandfunctionalanalysisofmetagenomes',\n",
        "  'AQUAautomatedqualityimprovementformultiplesequencealignments',\n",
        "  'RhodopsinlikeproteinfromthepurplemembraneofHalobacteriumhalobium',\n",
        "  'CompletegenomesequenceofCandidatusPuniceispirillummarinumIMCC1322arepresentativeoftheSAR116cladeintheAlphaproteobacteria',\n",
        "  'FastTreecomputinglargeminimumevolutiontreeswithprolesinsteadofadistancematrix',\n",
        "  'Dropsizesodalakestransientmicrobialhabitatsonasaltsecretingdeserttree',\n",
        "  'TheSorcererIIGlobalOceanSamplingexpeditionIThenorthwestAtlanticthroughtheeasterntropicalPacic',\n",
        "  'AdaptationandspectraltuningindivergentmarineproteorhodopsinsfromtheeasternMediterraneanandtheSargassoSeas',\n",
        "  'Actinorhodopsinsproteorhodopsinlikegenesequencesfoundpredominantlyinnonmarineenvironments',\n",
        "  'ActinorhodopsingenesdiscoveredindiversefreshwaterhabitatsandamongcultivatedfreshwaterActinobacteria',\n",
        "  'Themultitalentedmicrobialsensoryrhodopsins',\n",
        "  'Proteorhodopsingenesarewidelydistributedamongdivergentbacterialtaxa',\n",
        "  'AssessingdiversityandbiogeographyofaerobicanoxygenicphototrophicbacteriainsurfacewatersoftheAtlanticandPacicOceansusingtheGlobalOceanSamplingexpeditionmetagenomes']}"
       ]
      }
     ],
     "prompt_number": 16
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "df[:5]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>0</th>\n",
        "      <th>1</th>\n",
        "      <th>2</th>\n",
        "      <th>3</th>\n",
        "      <th>4</th>\n",
        "      <th>5</th>\n",
        "      <th>6</th>\n",
        "      <th>7</th>\n",
        "      <th>8</th>\n",
        "      <th>9</th>\n",
        "      <th>...</th>\n",
        "      <th>30</th>\n",
        "      <th>31</th>\n",
        "      <th>32</th>\n",
        "      <th>33</th>\n",
        "      <th>34</th>\n",
        "      <th>35</th>\n",
        "      <th>36</th>\n",
        "      <th>37</th>\n",
        "      <th>38</th>\n",
        "      <th>39</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>0</th>\n",
        "      <td> 0.251908</td>\n",
        "      <td> 0.251908</td>\n",
        "      <td> 0.244275</td>\n",
        "      <td> 0.229008</td>\n",
        "      <td> 0.221374</td>\n",
        "      <td> 0.221374</td>\n",
        "      <td> 0.244275</td>\n",
        "      <td> 0.229008</td>\n",
        "      <td> 0.236641</td>\n",
        "      <td> 0.259542</td>\n",
        "      <td>...</td>\n",
        "      <td> 0.129771</td>\n",
        "      <td> 0.244275</td>\n",
        "      <td> 0.129771</td>\n",
        "      <td> 0.190840</td>\n",
        "      <td> 0.190840</td>\n",
        "      <td> 0.198473</td>\n",
        "      <td> 0.152672</td>\n",
        "      <td> 0.229008</td>\n",
        "      <td> 0.236641</td>\n",
        "      <td> 0.206107</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1</th>\n",
        "      <td> 0.201550</td>\n",
        "      <td> 0.232558</td>\n",
        "      <td> 0.201550</td>\n",
        "      <td> 0.240310</td>\n",
        "      <td> 0.193798</td>\n",
        "      <td> 0.193798</td>\n",
        "      <td> 0.217054</td>\n",
        "      <td> 0.263566</td>\n",
        "      <td> 0.209302</td>\n",
        "      <td> 0.232558</td>\n",
        "      <td>...</td>\n",
        "      <td> 0.139535</td>\n",
        "      <td> 0.240310</td>\n",
        "      <td> 0.116279</td>\n",
        "      <td> 0.178295</td>\n",
        "      <td> 0.186047</td>\n",
        "      <td> 0.201550</td>\n",
        "      <td> 0.124031</td>\n",
        "      <td> 0.193798</td>\n",
        "      <td> 0.240310</td>\n",
        "      <td> 0.155039</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2</th>\n",
        "      <td> 0.235294</td>\n",
        "      <td> 0.247525</td>\n",
        "      <td> 0.233333</td>\n",
        "      <td> 0.234568</td>\n",
        "      <td> 0.250000</td>\n",
        "      <td> 0.125000</td>\n",
        "      <td> 0.240000</td>\n",
        "      <td> 0.246753</td>\n",
        "      <td> 0.211538</td>\n",
        "      <td> 0.252632</td>\n",
        "      <td>...</td>\n",
        "      <td> 0.250000</td>\n",
        "      <td> 0.282353</td>\n",
        "      <td> 0.187500</td>\n",
        "      <td> 0.245902</td>\n",
        "      <td> 0.166667</td>\n",
        "      <td> 0.173077</td>\n",
        "      <td> 0.208333</td>\n",
        "      <td> 0.185185</td>\n",
        "      <td> 0.239437</td>\n",
        "      <td> 0.252033</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3</th>\n",
        "      <td> 0.191176</td>\n",
        "      <td> 0.207921</td>\n",
        "      <td> 0.200000</td>\n",
        "      <td> 0.197531</td>\n",
        "      <td> 0.215909</td>\n",
        "      <td> 0.183673</td>\n",
        "      <td> 0.226667</td>\n",
        "      <td> 0.233766</td>\n",
        "      <td> 0.230769</td>\n",
        "      <td> 0.221053</td>\n",
        "      <td>...</td>\n",
        "      <td> 0.204082</td>\n",
        "      <td> 0.211765</td>\n",
        "      <td> 0.142857</td>\n",
        "      <td> 0.180328</td>\n",
        "      <td> 0.142857</td>\n",
        "      <td> 0.153846</td>\n",
        "      <td> 0.285714</td>\n",
        "      <td> 0.185185</td>\n",
        "      <td> 0.183099</td>\n",
        "      <td> 0.195122</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4</th>\n",
        "      <td> 0.242105</td>\n",
        "      <td> 0.207921</td>\n",
        "      <td> 0.221053</td>\n",
        "      <td> 0.200000</td>\n",
        "      <td> 0.210526</td>\n",
        "      <td> 0.242105</td>\n",
        "      <td> 0.273684</td>\n",
        "      <td> 0.221053</td>\n",
        "      <td> 0.221154</td>\n",
        "      <td> 0.189474</td>\n",
        "      <td>...</td>\n",
        "      <td> 0.168421</td>\n",
        "      <td> 0.200000</td>\n",
        "      <td> 0.136842</td>\n",
        "      <td> 0.200000</td>\n",
        "      <td> 0.210526</td>\n",
        "      <td> 0.210526</td>\n",
        "      <td> 0.189474</td>\n",
        "      <td> 0.231579</td>\n",
        "      <td> 0.263158</td>\n",
        "      <td> 0.235772</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>5 rows \u00d7 40 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 20,
       "text": [
        "         0         1         2         3         4         5         6   \\\n",
        "0  0.251908  0.251908  0.244275  0.229008  0.221374  0.221374  0.244275   \n",
        "1  0.201550  0.232558  0.201550  0.240310  0.193798  0.193798  0.217054   \n",
        "2  0.235294  0.247525  0.233333  0.234568  0.250000  0.125000  0.240000   \n",
        "3  0.191176  0.207921  0.200000  0.197531  0.215909  0.183673  0.226667   \n",
        "4  0.242105  0.207921  0.221053  0.200000  0.210526  0.242105  0.273684   \n",
        "\n",
        "         7         8         9     ...           30        31        32  \\\n",
        "0  0.229008  0.236641  0.259542    ...     0.129771  0.244275  0.129771   \n",
        "1  0.263566  0.209302  0.232558    ...     0.139535  0.240310  0.116279   \n",
        "2  0.246753  0.211538  0.252632    ...     0.250000  0.282353  0.187500   \n",
        "3  0.233766  0.230769  0.221053    ...     0.204082  0.211765  0.142857   \n",
        "4  0.221053  0.221154  0.189474    ...     0.168421  0.200000  0.136842   \n",
        "\n",
        "         33        34        35        36        37        38        39  \n",
        "0  0.190840  0.190840  0.198473  0.152672  0.229008  0.236641  0.206107  \n",
        "1  0.178295  0.186047  0.201550  0.124031  0.193798  0.240310  0.155039  \n",
        "2  0.245902  0.166667  0.173077  0.208333  0.185185  0.239437  0.252033  \n",
        "3  0.180328  0.142857  0.153846  0.285714  0.185185  0.183099  0.195122  \n",
        "4  0.200000  0.210526  0.210526  0.189474  0.231579  0.263158  0.235772  \n",
        "\n",
        "[5 rows x 40 columns]"
       ]
      }
     ],
     "prompt_number": 20
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for i in range(len(df.columns)):\n",
      "    for j in range(len(df.index)):\n",
      "        if df[i][j] > 0.5:\n",
      "            print df[i][j]\n",
      "            print filelist[\"1.pdfx\"][j]\n",
      "            print filelist[\"2.pdfx\"][i]\n",
      "            print 1-compare_dist(filelist[\"1.pdfx\"][j], filelist[\"2.pdfx\"][i])"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "0.516853932584\n",
        "Evolutionofreproductiveeffortinametapopulationwithlocalextinctionsandecologicalsuccession\n",
        "Neutralgeneticdiversityinametapopulationwithrecurrentlocalextinctionandrecolonization\n",
        "0.516853932584\n",
        "0.516853932584\n"
       ]
      },
      {
       "ename": "IndexError",
       "evalue": "list index out of range",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-19-175468b99196>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      3\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mj\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m>\u001b[0m \u001b[1;36m0.5\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m             \u001b[1;32mprint\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mj\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 5\u001b[1;33m             \u001b[1;32mprint\u001b[0m \u001b[0mfilelist\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"1.pdfx\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mj\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      6\u001b[0m             \u001b[1;32mprint\u001b[0m \u001b[0mfilelist\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"2.pdfx\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      7\u001b[0m             \u001b[1;32mprint\u001b[0m \u001b[1;36m1\u001b[0m\u001b[1;33m-\u001b[0m\u001b[0mcompare_dist\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfilelist\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"1.pdfx\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mj\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfilelist\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"2.pdfx\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;31mIndexError\u001b[0m: list index out of range"
       ]
      }
     ],
     "prompt_number": 19
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "df[]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "help(distance.levenshtein)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Help on function levenshtein in module distance._levenshtein:\n",
        "\n",
        "levenshtein(seq1, seq2, max_dist=-1, normalized=False)\n",
        "    Compute the absolute Levenshtein distance between the two sequences\n",
        "    `seq1` and `seq2`.\n",
        "    \n",
        "    The Levenshtein distance is the minimum number of edit operations necessary\n",
        "    for transforming one sequence into the other. The edit operations allowed are:\n",
        "    \n",
        "            * deletion:     ABC -> BC, AC, AB\n",
        "            * insertion:    ABC -> ABCD, EABC, AEBC..\n",
        "            * substitution: ABC -> ABE, ADC, FBC..\n",
        "    \n",
        "    The `max_dist` parameter controls at which moment we should stop computing the\n",
        "    distance between the provided sequences. If it is a negative integer, the\n",
        "    distance will be computed until the sequences are exhausted; otherwise, the\n",
        "    computation will stop at the moment the calculated distance is higher than\n",
        "    `max_dist`, and then return -1. For example:\n",
        "    \n",
        "            >>> levenshtein(\"abc\", \"abcd\", max_dist=1)  # dist = 1\n",
        "            1\n",
        "            >>> levenshtein(\"abc\", \"abcde\", max_dist=1) # dist = 2\n",
        "            -1\n",
        "    \n",
        "    This can be a time saver if you're not interested in the exact distance, but\n",
        "    only need to check if the distance between the given sequences is below a\n",
        "    given threshold.\n",
        "    \n",
        "    The `normalized` parameter is here for backward compatibility; providing\n",
        "    it will result in a call to `nlevenshtein`, which should be used directly\n",
        "    instead.\n",
        "\n"
       ]
      }
     ],
     "prompt_number": 302
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}